authentication/account/worker/user_deactivation_notification_worker_test.go

```diff
@@ -0,0 +1,127 @@
+	appservicemock "github.com/fabric8-services/fabric8-auth/test/generated/application/service"
+	accountservicemock "github.com/fabric8-services/fabric8-auth/test/generated/authentication/account/service"
+	baseworker "github.com/fabric8-services/fabric8-auth/worker/repository"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type UserDeactivationNotificationWorkerTest struct {
+	gormtestsupport.DBTestSuite
+}
+
+func TestUserDeactivationNotificationWorker(t *testing.T) {
+	suite.Run(t, &UserDeactivationNotificationWorkerTest{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *UserDeactivationNotificationWorkerTest) TestNotifyUsers() {
+	// given
+	config := accountservicemock.NewUserServiceConfigurationMock(s.T())
+	config.GetUserDeactivationFetchLimitFunc = func() int {
+		return 100
+	}
+	config.GetUserDeactivationInactivityPeriodDaysFunc = func() time.Duration {
+		return 30 * 24 * time.Hour // 31 days, ie, 7 days after notification
+	}
+	config.GetUserDeactivationInactivityNotificationPeriodDaysFunc = func() time.Duration {
+		return 20 * 24 * time.Hour // 24 days
+	}
+	config.GetPostDeactivationNotificationDelayMillisFunc = func() time.Duration {
+		return 5 * time.Millisecond
+	}
+	ctx := context.Background()
+	// yesterday := time.Now().Add(-1 * 24 * time.Hour)
+	ago40days := time.Now().Add(-40 * 24 * time.Hour) // 40 days since last activity and notified...
+
+	var notificationServiceMock *appservicemock.NotificationServiceMock
+	var app application.Application
+	s.SetupSubtest = func() {
+		notificationServiceMock = appservicemock.NewNotificationServiceMock(s.T())
+		notificationServiceMock.SendMessageAsyncFunc = func(ctx context.Context, msg notification.Message, options ...rest.HTTPClientOption) (r chan error, r1 error) {
+			return nil, nil
+		}
+		app = gormapplication.NewGormDB(s.DB, s.Configuration, s.Wrappers, factory.WithNotificationService(notificationServiceMock))
+	}
+
+	s.Run("one user to notify once", func() {
+		// given
+		user1 := s.Graph.CreateUser().User()
+		identity1 := user1.Identities[0]
+		identity1.LastActive = &ago40days
+		err := s.Application.Identities().Save(ctx, &identity1)
+		require.NoError(s.T(), err)
+		// start the worker with a 50ms ticker
+		w := worker.NewUserDeactivationNotificationWorker(context.Background(), app)
+		w.Start(time.Millisecond * 50)
+		defer w.Stop()
+		// wait a few cycles before checking the results
+		time.Sleep(time.Millisecond * 100)
+		// then load the user and check her deactivation notification status
+		result, err := s.Application.Identities().Load(context.Background(), identity1.ID)
+		require.NoError(s.T(), err)
+		assert.NotNil(s.T(), result.DeactivationNotification)
+		// notification only sent once to the user
+		assert.Equal(s.T(), uint64(1), notificationServiceMock.SendMessageAsyncCounter)
+	})
+
+	s.Run("multiple workers but only one working", func() {
```

Comment:
I think that you need to put it into a separated test (not subtest) to make the suite passing - the first subtest deactivates the user, so there is nothing left for the second one.